### PR TITLE
[CI] Enable complexity check for the Python code.

### PR DESCRIPTION
**Pull request by Cjkjvfnby**
It requires efforts to maintain low complexity but it's much easier than working with complex functions. 

This PR enables complexity checks and suppresses existing problems. 

I have set the complexity to 10.  Let's see how it will work.
https://en.wikipedia.org/wiki/Cyclomatic_complexity#Limiting_complexity_during_development

### CODE REVIEW

**Comment by Cjkjvfnby** [comment]
@Grummel7 what do you think?

File: default/python/AI/AIFleetMission.py
Comment (Grummel7) [inline default/python/AI/AIFleetMission.py]: This disables the warning for the whole file, see https://stackoverflow.com/questions/51127170/flake8-ignore-warnings-for-a-function

Note that I also found that when you put it in the first line of the function, you must write `# noqa: C901` or `# noqa: C901 flake8`. With flake8 before the noqa flake8 will ignore it.

File: default/python/tox.ini
Comment (Grummel7) [inline default/python/tox.ini:5]: To be honest, I do not like a hard limit of 10. During development I often check in more complex functions only to split them up once I figured out what the code should do. With AI development this is not always trivial.

Can we agree on 15 here? That doesn't stop us from aiming lower, but I don't want to keep adding noqas.

Btw. #4137 is still waiting for your approval.

File: default/python/AI/AIFleetMission.py
Comment (Cjkjvfnby) [reply default/python/AI/AIFleetMission.py]: Thank you.  Nice catch.

File: default/python/tox.ini
Comment (Cjkjvfnby) [reply default/python/tox.ini:5]: Lets try to keep it a 10 for a while.

If this number will lead to too many suppressions we will update it.  
Now we have 64 suppressions, and we could get back to this after we will have more than 70. 

During development, you could just commit with `-n` / `--no-verify`.
I usually try to commit as is and fix other warnings reported by pre-commit. 

When you finished you could run eigher `flake8` or `pre-commit run --all-files`

**Review by Grummel7 — APPROVED** [review]
Of all the functions I contributed to the AI, only one had a complexity of 12.
Go ahead.

**Pull request merged**
